docker/README.md

```diff
@@ -0,0 +1,311 @@
+    10. Generate Entropy!
+    
+2. gpg -a --export > ./docker/public.key
+
+3. gpg -a --export-secret-keys > ./docker/signing.key
+
+Keep in mind, signing.key must NEVER be accidentally committed to the repo. 
+
+To prevent MANY passphrase entries at random times, the gpg-agent is used to
+cache your signing key. This is done by running gpg-agent in the host, and passing
+the connection to each docker image. This will be done for you automatically by
+calling the `./docker/preload_key.bsh` script. This can be called manually
+before any other command just to get the pass phrase entry out of the way before
+you start running everything.
+
+GPG agent TTL is set to 5 hours. This should be plenty to build everything. If this is
+not good for you, set the GPG_MAX_CACHE and GPG_DEFAULT_CACHE environment variables
+(in seconds)
+
+### GPG capabilities by Distro ###
+
+Debian WILL work with 4096 bit RSA signing subkeys like [1] suggests, but will
+also work with 4096 bit RSA signing keys.
+
+CentOS will NOT work with subkeys[3]. CentOS 6 and 7 will work with 4096 bit RSA 
+signing keys
+
+CentOS 5 will NOT work with v4 signatures. The rpms will be so unrecognizable 
+that it can't even be installed with --nogpgcheck. It should work with RSA on v3.
+However, I could not get it to. It builds v3 correctly, but for some reason
+the GPG check fails for RSA. However `yum install --nogpgcheck` does work! CentOS 5
+will NOT work with 2048 bit DSA keys... I suspect 2048 is too big for it to
+fathom. CentOS 5 WILL work with 1024 bit DSA keys. Either sign it with the key
+instructions above and install it with the `yum install --nogpgcheck` OR
+create a NEW DSA key just for CentOS 5 and sign with that. 
+
+[1] https://www.digitalocean.com/community/tutorials/how-to-use-reprepro-for-a-secure-package-repository-on-ubuntu-14-04
+
+[2] https://iuscommunity.org/pages/CreatingAGPGKeyandSigningRPMs.html#exporting-the-public-gpg-key
+
+[3] http://www.redhat.com/archives/rpm-list/2006-November/msg00105.html
+
+### Testing the Repositories ###
+
+To test that all the OSes can download the rpm/debs, install, and run the tests
+again, run
+
+    ./test_dockers.bsh
+    
+(which is basically just `./docker/run_dockers.bsh ./docker/git-lfs-test_*`)
+
+Remember to set REPO_HOSTNAME if you changed it for `./docker/build_docker.bsh`
+This can also be used to run a local test (on `localhost:{Port Number}`, for
+example)
+
+An easy way to test the repositories locally, is to run them on a simple webserver such as
+
+    cd ./repos
+    python -m SimpleHTTPServer {Port number}
+
+or
+
+    cd ./repos
+    ruby -run -ehttpd . -p{Port Number}
+
+
+## Adding addition OSes ##
+
+To add another operating system, simply follow the already existing pattern, 
+and all the scripts will pick them up. A new Dockerfile should go in a directory
+named
+
+    ./docker/git-lfs_{OS NAME}_{OS VERSION #}
+    
+where **{OS NAME}** and **{OS VERSION #}** should not contain underscores (_).
```

Comment:
Can you put backticks around the `_`?  The markdown rendering is italicizing the rest of the text below.